hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +279,230 @@ default Table newHmsTable(String hmsTableOwner) {
+          throw new RuntimeException(
+              "Failed to acquire locks from metastore because the underlying metastore "
+                  + "table 'HIVE_LOCKS' does not exist. This can occur when using an embedded metastore which does not "
+                  + "support transactions. To fix this use an alternative metastore.",
+              e);
+        }
+
+        LOG.error(
+            "Cannot tell if commit to {}.{} succeeded, attempting to reconnect and check.",
+            database(),
+            table(),
+            e);
+        commitStatus = validateNewLocationAndReturnCommitStatus(metadata, newMetadataLocation);
+
+        switch (commitStatus) {
+          case SUCCESS:
+            break;
+          case FAILURE:
+            throw e;
+          case UNKNOWN:
+            throw new CommitStateUnknownException(e);
+        }
+      }
+    } catch (TException e) {
+      throw new RuntimeException(
+          String.format("Metastore operation failed for %s.%s", database(), table()), e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted during commit", e);
+
+    } catch (LockException e) {
+      throw new CommitFailedException(e);
+    } finally {
+      cleanupMetadataAndUnlock(lock, io, commitStatus, newMetadataLocation);
+    }
+  }
+
+  default String fullName() {
+    return catalogName() + "." + database() + "." + table();
+  }
+
+  default StorageDescriptor storageDescriptor(BaseMetadata metadata, boolean hiveEngineEnabled) {
+    return HiveOperationsBase.storageDescriptor(
+        metadata.schema(), metadata.location(), hiveEngineEnabled);
+  }
+
+  default Set<String> obsoleteProps(Configuration conf, BaseMetadata base, BaseMetadata metadata) {
+    Set<String> obsoleteProps = Sets.newHashSet();
+    if (base != null) {
+      obsoleteProps =
+          base.properties().keySet().stream()
+              .filter(key -> !metadata.properties().containsKey(key))
+              .collect(Collectors.toSet());
+    }
+
+    if (!conf.getBoolean(ConfigProperties.KEEP_HIVE_STATS, false)) {
+      obsoleteProps.add(StatsSetupConst.COLUMN_STATS_ACCURATE);
+    }
+
+    return obsoleteProps;
+  }
+
+  /**
+   * Returns if the hive engine related values should be enabled on the table, or not.
+   *
+   * <p>The decision is made like this:
+   *
+   * <ol>
+   *   <li>Table property value {@link TableProperties#ENGINE_HIVE_ENABLED}
+   *   <li>If the table property is not set then check the hive-site.xml property value {@link
+   *       ConfigProperties#ENGINE_HIVE_ENABLED}
+   *   <li>If none of the above is enabled then use the default value {@link
+   *       TableProperties#ENGINE_HIVE_ENABLED_DEFAULT}
+   * </ol>
+   *
+   * @param metadata Table metadata to use
+   * @return if the hive engine related values should be enabled or not
+   */
+  default boolean hiveEngineEnabled(Configuration conf, BaseMetadata metadata) {
```

Comment:
why do we pass `BaseMetadata` here if this method is only applicable for tables? `TableProperties.ENGINE_HIVE_ENABLED` won't be set on a view

Reply:
This has been changed with default value as false as per recommendation from @szehon-ho 
https://github.com/apache/iceberg/pull/9852#discussion_r1518485623